### PR TITLE
Union support

### PR DESCRIPTION
**Pull request by 7sharp9**

### CODE REVIEW

File: src/Falanx.Machinery/Expr.fs [hunk 1]
```diff
@@ -16,13 +16,18 @@ namespace Falanx.Machinery
...
               .Replace("Collections.Generic.","")
               .Replace("Falanx.JsonCodec.", "")
-              .Replace("Newtonsoft.Json.Linq.", "") 
+              .Replace("Newtonsoft.Json.Linq.", "")
+              .Replace("`4", "")
               .Replace("`2", "")
               .Replace("`1", "")
-              .Replace("FSharp", "")
+              .Replace("FSharp.Collections.", "")
+              .Replace("Fleece.Newtonsoft+", "")
```
Comment (enricosada) [inline src/Falanx.Machinery/Expr.fs]: is `Fleece.Newtonsoft+` right or should be `Fleece.Newtonsoft.`?

File: src/Falanx.Machinery/Reflection.fs [hunk 2]
```diff
@@ -121,10 +122,85 @@ module Reflection =
               get typ 
             else typ
             
+        let maxTuple = 8
+        
+        // Which field holds the nested tuple?
+        let tupleEncField = maxTuple - 1
+        
+        let simpleTupleNames = [|
+            "Tuple`1";      "Tuple`2";      "Tuple`3";
```
Comment (enricosada) [inline src/Falanx.Machinery/Reflection.fs]: maybe

```fsharp
for i in 1..8 do
   yield sprintf "Tuple`%i" i
   yield sprintf "ValueTuple`%i" i
```

File: src/Falanx.Machinery/Expr.fs (hunk 1 above)
Comment (7sharp9) [reply src/Falanx.Machinery/Expr.fs]: Yeah this was to get rid of nested classes

File: src/Falanx.Machinery/Reflection.fs (hunk 2 above)
Comment (7sharp9) [reply src/Falanx.Machinery/Reflection.fs]: This is code ripped from the compiler so...

Comment (7sharp9) [reply src/Falanx.Machinery/Reflection.fs]: Im going to leave as it is for now, the code generated form an array comprehension is about 5x more than the constant version.

**Pull request merged**